### DISCUSSION TITLE
Settings Configuration Persistence

### DISCUSSION THREAD

**Discussion by fire1ce** [Selbi182/SpotifyBigPicture#26] [Ideas]
Hey @Selbi182! 
First of all. Thanks for your great work.
As i can see from the source all the app is bundled to jar file including the setting changes made by user. 
It will present an issue with updating the app / container since the all setting will be lost since the Jar file will be replaced entirely.
If i'm wrong, please correct me, but i think personal settings should be saved separate folder so when the container / app will be updated only the binary  (jar) will be replaced without losing the Personal Settings made by user.

**Comment by Selbi182** [comment]
Hey, thanks for the kind words! :)

The settings are actually saved inside your browser, not anywhere in the JAR file (those are read-only anyway afaik). You can safely replace the old JAR with a newer version.

However, do note that between versions I sometimes add/delete/rename certain settings, so it's possible that inconsistencies might occur. Therefore, if you do update to a new version, you will be greeted by a prompt asking you to reset your settings to avoid inconsistencies. This isn't mandatory but recommended if you want to be on the safe side.

**Reply by fire1ce** [reply]
@Selbi182 Thanks for the very fast replay. 
Sorry my mistake, didn't look at the Local storage. 
May i suggest to add a export/import settings feature? 
here's some info about the it:
https://stackoverflow.com/questions/13335967/export-data-in-localstorage-for-later-re-import

**Reply by Selbi182** [reply]
I suppose I could write an import/export feature. It never really crossed my mind but it shouldn't be too difficult.
